### PR TITLE
Simple HEAD request

### PR DESCRIPTION
**Pull request by Himalee**
Given: Executing a HEAD request to /simple_get
When: I make a HEAD request to "/simple_get"
Then: My response should have status code 200
And: My response should have no body

### CODE REVIEW

File: src/main/java/WebServer.java [hunk 1]
```diff
@@ -27,9 +28,15 @@ public void start(int port) throws IOException {
...
-        if (request().contains(SIMPLE_GET_REQUEST)) {
-            responseBuilder = new ResponseBuilder();
-            responseHandler.respond(output, responseBuilder.okayWithEmptyBody());
+        responseBuilder = new ResponseBuilder();
+        String request = request();
+        switch(request) {
+            case SIMPLE_GET_REQUEST:
+                responseHandler.respond(output, responseBuilder.okayWithEmptyBody());
+                break;
+            case SIMPLE_HEAD_REQUEST:
```
Comment (ssunday) [inline src/main/java/WebServer.java]: nice

File: src/main/java/WebServer.java [hunk 2]
```diff
@@ -27,9 +28,15 @@ public void start(int port) throws IOException {
     private void respond() throws IOException {
         serverCommunicationChannel = serverSocketManager.acceptConnection();
         OutputStream output = serverCommunicationChannel.getOutputStream();
-        if (request().contains(SIMPLE_GET_REQUEST)) {
-            responseBuilder = new ResponseBuilder();
-            responseHandler.respond(output, responseBuilder.okayWithEmptyBody());
+        responseBuilder = new ResponseBuilder();
+        String request = request();
+        switch(request) {
+            case SIMPLE_GET_REQUEST:
```
Comment (ssunday) [inline src/main/java/WebServer.java]: what would it take to be able to respond to HEAD from whatever route?

Comment (Himalee) [reply src/main/java/WebServer.java]: Checking if the request contains `HEAD`? e.g. `if (request.contains("HEAD"))`

Comment (ssunday) [reply src/main/java/WebServer.java]: 👍

File: src/test/java/ResponseBuilderTest.java [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+        responseBuilder = new ResponseBuilder();
+    }
     @Test
     public void buildResponseWithStatusCode200AndEmptyBody() {
-        ResponseBuilder responseBuilder = new ResponseBuilder();
         Assert.assertEquals("HTTP/1.1 200 OK\r\n\r\n", responseBuilder.okayWithEmptyBody());
     }
+
+    @Test
+    public void buildResponseWithStatusCode20AndNoBody() {
```
Comment (amb8l) [inline src/test/java/ResponseBuilderTest.java]: There is a typo in `StatusCode20`, I think you mean `StatusCode200`.

What is the difference between `responseBuilder.okayWithEmptyBody()` and `responseBuilder.okayWithNoBody()`? To me, it looks like `responseBuilder.okayWithEmptyBody() == responseBuilder.okayWithNoBody()`

Comment (Himalee) [reply src/test/java/ResponseBuilderTest.java]: They are the same, the difference is one has no body (`HEAD` request) and one has an empty body represented by an empty string (`GET` request) - I wasn't sure whether there needed to be a distinction between the two requests. If they are the same, then I guess I should remove the duplication!

File: src/test/java/WebServerTest.java [hunk 4]
```diff
@@ -11,25 +10,33 @@
...
+        CommunicationChannel mockServerCommunicationChannel = new MockServerCommunicationChannel(request);
         mockServerSocketManager = new MockServerSocketManager(mockServerCommunicationChannel);
         webServer = new WebServer(mockServerSocketManager, requestHandler, responseHandler, mockWebServerStatus);
         webServer.start(port);
     }
 
     @Test
-    public void startWebServerListenForConnections() {
+    public void startWebServerListenForConnections() throws IOException {
+        setUp("GET /simple_get HTTP/1.1");
```
Comment (amb8l) [inline src/test/java/WebServerTest.java]: I like that setup that is not necessary for the tests are abstracted away into this `setUp` method. However, I found that I was having to go and look at the method to figure out what the test was doing.

I wonder if a name change and perhaps a non-`void` return type might help make this clearer?

**Pull request merged**